Enable inspector tests for ChakraCore

kfarnung opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-guidelines)

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->
build, test

kfarnung committed:
test: enable inspector tests

Enabled tests for inspector and updated them to account for
differences between V8 and ChakraCore.

PR-URL: https://github.com/nodejs/node-chakracore/pull/247
Reviewed-By: Hitesh Kanwathirtha <hiteshk@microsoft.com>
Reviewed-By: Kunal Pathak <Kunal.Pathak@microsoft.com>

kfarnung committed:
build,src: enable intl for chakracore

* Enabled intl support to get UTF-8/UTF-16 conversion for inspector.
  This is a temporary move before actual intl support is enabled.
* Removed workaround for UTF-8 conversion functions that weren't
  available before ICU was enabled.

PR-URL: https://github.com/nodejs/node-chakracore/pull/247
Reviewed-By: Hitesh Kanwathirtha <hiteshk@microsoft.com>
Reviewed-By: Kunal Pathak <Kunal.Pathak@microsoft.com>

digitalinfinity commented on test/inspector/inspector.status:
```diff
@@ -9,4 +9,3 @@ prefix inspector
 [$system==win32]
 
 [$jsEngine==chakracore]
```
Why not remove this section altogether and have it match upstream?

kfarnung replied:
Easy enough, I'll do that.

digitalinfinity commented on test/parallel/test-whatwg-url-setters.js:
```diff
@@ -8,7 +8,7 @@ const { test, assert_equals } = common.WPT;
     path.join(common.fixturesDir, 'url-setter-tests-additional.js'));
 
-if (!common.hasIntl) {
+if (!common.hasIntl || common.isChakraEngine) {
```
Isn't this too broad? Like Intl should be available on Windows, no?

kfarnung replied:
I can take another look, but this was an issue on Windows in my testing.

kfarnung replied:
Odd, here's the failure on see on Windows:

```
=== release test-whatwg-url-setters ===
Path: parallel/test-whatwg-url-setters
TypeError: Object doesn't support property or method 'ToString'
   at href.set (internal/url.js:307:7)
   at Anonymous function (E:\GitHub\node-chakracore\test\parallel\test-whatwg-url-setters.js:121:27)
   at _tryBlock (assert.js:503:5)
   at _throws (assert.js:522:3)
   at throws (assert.js:552:3)
   at Anonymous function (E:\GitHub\node-chakracore\test\parallel\test-whatwg-url-setters.js:121:7)
   at Module.prototype._compile (module.js:582:5)
   at Module._extensions[.js] (module.js:593:3)
   at Module.prototype.load (module.js:516:3)
   at tryModuleLoad (module.js:466:5)
Command: E:\GitHub\node-chakracore\Release\node.exe E:\GitHub\node-chakracore\test\parallel\test-whatwg-url-setters.js
```

I'll investigate further.

kfarnung committed:
chakrashim: fixing jslint issues

A new rule was added to enforce function declaration style.

PR-URL: https://github.com/nodejs/node-chakracore/pull/247
Reviewed-By: Hitesh Kanwathirtha <hiteshk@microsoft.com>
Reviewed-By: Kunal Pathak <Kunal.Pathak@microsoft.com>

kfarnung commented on src/inspector_io.cc:
```diff
@@ -1,476 +1,445 @@
-#include "inspector_io.h"
```
This is the result of `git checkout nodejs/master src/inspector_io.cc`.  At some point the line endings must have gotten messed up.  The diff here just removes the UTF-8/UTF-16 conversion workarounds that I added for inspector.
